docs/index.md

```diff
@@ -1,3 +1,134 @@
+purpose of the [Roc Toolkit](https://roc-streaming.org/toolkit/docs/about_project/overview.html) (further
+referred to as `roc`) is to implement real-time audio streaming over network.
+
+## Entities
+
+Here are the top-level components:
+
+**host**
+:   Computer where the `roc` instance is currently running on. Computer platform may be on of:
+
+    - PC (personal computer, usually a desktop device with monitor or a similar output device),
+    - Android,
+    - iOS (future plans),
+    - embedded (is actually the same as PC, but usually doesn't have a monitor).
+
+**rocd**
+:   Server component used for stream management and auto discovery over network. Contains a library and a
+    binary that wraps it in a daemon. A stand-alone library is needed for mobile versions with their specific
+    platform limitations (programming language, permission models, ...).
+
+**roc-cast**
+:   Component used for two primary purposes:
+
+    - Provide UI for user interaction with rocd.
+    - On Android (and — possibly — iOS) devices it implements a wrapper application with a so-called main
+      activity context to allow interaction between non-Java libraries (C/C++/...) with mobile sound devices.
+
+**roc-toolkit**
+:   The part that provides some or all of the following components: sound device IO, pipeline, network IO. May
+    be used in binary (MVP) or library forms.
+
+## Usage cases
+
+Here `roc` may be used as a binary only as it doesn't currently implement C API for all 3 components:
+
+![](./assets/dia/simple.svg)
+/// caption
+Simple case: `roc` implements all steps between input and output sound devices
+///
+
+<!-- TODO: add more use case examples here -->
+
+![](./assets/dia/general.svg)
+/// caption
+...
+///
+
+## APIs
+
+`roc-toolkit` may be used in [different ways](https://roc-streaming.org/toolkit/docs/api/reference.html):
+
+![](./assets/dia/roc-toolkit-apis-1.svg)
+/// caption
+as a library for pipeline only (`roc_{sender,receiver}_{encoder,decoder}*`)
+///
+
+![](./assets/dia/roc-toolkit-apis-2.svg)
+/// caption
+as a library for pipeline and network components (`roc_{sender,receiver)*`)
+///
+
+![](./assets/dia/roc-toolkit-apis-3.svg)
+/// caption
+as a library for all three components (`roc_transceiver*`; not implemented)
+///
+
+All three components — `sndio`, `pipeline`, and `netio` — are configured:
+
+| Component | via CLI args | via C API |
+| ---:      | :---:        | :---:     |
+| sndio     | ✅           | ❌        |
+| pipeline  | ✅           | ✅        |
+| netio     | ✅           | ✅        |
+/// caption
+///
+
+## Implementation
+
+`roc-cast` will provide two interfaces:
+- native UI (Flutter) and
+- web UI.
+
+The latter may be used on embedded systems
+
+User configures `roc` (sound devices, network, pipeline) via native `roc-cast` client (primary way) or
+directly by interacting with `rocd` by HTTP API. On mobile platforms it is always `roc-cast` client. On
+embeded platforms the default way is HTTP client.
```

Comment:
I think if user uses roc-cast, it should never use rocd directly (unless debugging). roc-cast will likely have its own logic on top of rocd and I don't think we should allow/support bypassing that logic.

Reply:
removed this